prusti/src/ide_helper/compiler_info.rs

```diff
@@ -0,0 +1,128 @@
+use super::{call_finder, query_signature};
+use prusti_interface::{environment::Environment, specs::typed};
+use prusti_rustc_interface::{
+    hir::def_id::DefId,
+    span::{source_map::SourceMap, Span},
+};
+use prusti_viper::ide::vsc_span::VscSpan;
+use serde::{ser::SerializeStruct, Serialize};
+
+/// This struct will be passed to prusti-assistant containing information
+/// about the program that is currently being verified
+#[derive(Serialize)]
+pub struct IdeInfo {
+    procedure_defs: Vec<ProcDef>,
+    function_calls: Vec<ProcDef>,
+    queried_source: Option<String>,
+    // additionally this will contain:
+    // function_calls:
+    // ... we'll see
+}
+
+impl IdeInfo {
+    pub fn collect(
+        env: &Environment<'_>,
+        procedures: &Vec<DefId>,
+        def_spec: &typed::DefSpecificationMap,
+    ) -> Self {
+        let procs = collect_procedures(env, procedures, def_spec);
+        let source_map = env.tcx().sess.source_map();
+        let fncalls: Vec<ProcDef> = collect_fncalls(env)
+            .into_iter()
+            .map(|(name, defid, sp)| ProcDef {
+                name,
+                defid,
+                span: VscSpan::from_span(&sp, source_map).unwrap(),
+            })
+            .collect();
+
+        // For declaring external specifications:
+        let queried_source = query_signature::collect_queried_signature(env.tcx(), &fncalls);
+        Self {
+            procedure_defs: procs,
+            function_calls: fncalls,
+            queried_source,
+        }
+    }
+}
+
+/// A struct that contains either a reference to a procedure that can be verified
+/// (for selective verification) or a function call (so a user can query
+/// external_spec blocks for it). The name contains the defpath.
+pub struct ProcDef {
+    pub name: String,
+    pub defid: DefId,
+    pub span: VscSpan,
+}
+
+impl Serialize for ProcDef {
```

Comment:
You can add [`#[serde(skip)]`](https://serde.rs/field-attrs.html#skip) to the `defid` field to skip it, instead of manually implementing the trait.